data/awsrds/v0/main.go

```diff
@@ -0,0 +1,448 @@
+	sqlStatement += "ON DUPLICATE KEY UPDATE " + strings.Join(updateColumns, ", ")
+
+	// Prepare and execute the statement using NamedExec
+	_, err := client.NamedExec(sqlStatement, values)
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func SelectUser(client *sqlx.DB, tableName *string, criteria *structpb.Struct, from int, to int) (*structpb.ListValue, error) {
+	// Begin constructing SQL statement
+	sqlStatement := "SELECT "
+	var where []string
+	var columns []string
+
+	for criteriaKey, criteriaValue := range criteria.GetFields() {
+		if criteriaValue.GetKind() != nil {
+			switch criteriaValue.Kind.(type) {
+			case *structpb.Value_StringValue:
+				// If the value is a string, quote it in the SQL statement
+				where = append(where, fmt.Sprintf("%s = '%s'", criteriaKey, criteriaValue.GetStringValue()))
+			case *structpb.Value_NumberValue, *structpb.Value_BoolValue:
+				// If the value is a number or bool, use it directly without quotes
+				where = append(where, fmt.Sprintf("%s = %v", criteriaKey, criteriaValue.AsInterface()))
+			}
+		}
+
+		columns = append(columns, criteriaKey)
+	}
+
+	var notAll string
+	if to == 0 && from == 0 {
+		notAll = ""
+	} else {
+		notAll = " LIMIT " + strconv.Itoa(to-from+1) + " OFFSET " + strconv.Itoa(from-1)
+	}
+
+	sqlStatement += strings.Join(columns, ", ")
+
+	sqlStatement += " FROM " + *tableName
+	if len(where) > 0 {
+		sqlStatement += " WHERE " + strings.Join(where, " AND ")
+	}
+	sqlStatement += notAll
+
+	// Prepare and execute the statement
+	rows, err := client.Queryx(sqlStatement)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	// Prepare the result ListValue
+	result := &structpb.ListValue{}
+
+	// Iterate over the rows
+	for rows.Next() {
+		// Create a map to hold the row data
+		rowMap := make(map[string]any)
+
+		// Load the row data into the map
+		err := rows.MapScan(rowMap)
+		if err != nil {
+			return nil, fmt.Errorf("failed to scan row: %v", err)
+		}
+
+		// Convert each value in the map to the appropriate type
+		for key, value := range rowMap {
+			switch v := value.(type) {
+			case []byte:
+				// Convert byte slices to strings
+				rowMap[key] = string(v)
+			}
+		}
+
+		// Convert the map to a structpb.Struct
+		structValue, err := structpb.NewStruct(rowMap)
+		if err != nil {
+			return nil, fmt.Errorf("failed to convert map to structpb.Struct: %v", err)
+		}
+
+		// Convert the structpb.Struct to a structpb.Value
+		structValuePb := structpb.NewStructValue(structValue)
+
+		// Add the structpb.Value to the result ListValue
+		result.Values = append(result.Values, structValuePb)
+	}
+
+	// Check for errors after iterating over rows
+	if err = rows.Err(); err != nil {
+		return nil, fmt.Errorf("error occurred during row iteration: %v", err)
+	}
+
+	return result, nil
+}
+
+func DeleteUser(client *sqlx.DB, tableName *string, criteria *structpb.Struct) error {
```

Comment:
Important
It seems to delete records rather than a record.
So, please clarify this point in the document.